recipe/deactivate-gcc.sh

```diff
@@ -88,13 +88,15 @@ if [ "${CONDA_BUILD:-0}" = "1" ]; then
   CPPFLAGS_USED="@CPPFLAGS@ -I${PREFIX}/include"
   DEBUG_CPPFLAGS_USED="@DEBUG_CPPFLAGS@ -I${PREFIX}/include"
   CMAKE_PREFIX_PATH_USED="${CMAKE_PREFIX_PATH}:${PREFIX}:${BUILD_PREFIX}/${HOST}/sysroot/usr"
+  CONDA_BUILD_SYSROOT_USED="${PREFIX}/@CHOST@/sysroot"
```

Comment:
This should be `BUILD_PREFIX` as that's where the compiler is.

Reply:
hmm, this also means that `test -d ${CONDA_BUILD_SYSROOT}` should be removed from meta.yaml?

Reply:
```suggestion
  CONDA_BUILD_SYSROOT_USED="${BUILD_PREFIX:-${CONDA_PREFIX}}/@CHOST@/sysroot"
```

Reply:
@isuruf are you sure? When running `${CC} --print-sysroot` from `meta.yaml`, the result is
```
$PREFIX/bin/../x86_64-conda_cos6-linux-gnu/sysroot
```
Also, only with the option
```
CONDA_BUILD_SYSROOT_USED="${PREFIX}/@CHOST@/sysroot"
```
the tests are passed. In other cases with `BUILD_PREFIX` or `CONDA_PREFIX`, the tests fail.

Reply:
Okay. Then you should use `${BUILD_PREFIX:-${PREFIX}}`. The issue will pop up in other feedstocks, where the compiler gets installed to the `BUILD_PREFIX` instead of `PREFIX`

Reply:
Ok, this makes sense.

Since `BUILD_PREFIX` is always defined when `CONDA_BUILD=1`, then `PREFIX` fallback seems unnecessary.

Also, the `test -d CONDA_BUILD_SYSROOT` will now report that the directory does not exist.
What is the environment when running the meta.yaml tests? Should we activate gcc in the meta.yaml tests before executing `test -d CONDA_BUILD_SYSROOT` ?

Reply:
Can you print the values of `BUILD_PREFIX`, `PREFIX` and `CONDA_PREFIX` in the activate script?

Reply:
Yes, from the activate-gcc script:
```
PREFIX=$PREFIX
CONDA_PREFIX=$PREFIX
BUILD_PREFIX=/home/pearu/miniconda3/conda-bld/ctng-compiler-activation_1573773024119/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh
```

Reply:
Thanks. Can you also print the CONDA_BUILD_STATE variable?

Reply:
```
CONDA_BUILD_STATE=TEST
```